etna/transforms/decomposition/base_change_points.py

```diff
@@ -0,0 +1,115 @@
+from abc import ABC
+from abc import abstractmethod
+from typing import List
+from typing import Tuple
+from typing import Type
+
+import pandas as pd
+from ruptures.base import BaseEstimator
+from ruptures.costs import CostLinear
+from sklearn.base import RegressorMixin
+
+TTimestampInterval = Tuple[pd.Timestamp, pd.Timestamp]
+TDetrendModel = Type[RegressorMixin]
+
+
+class BaseChangePointsModelAdapter(ABC):
+    """BaseChangePointsModelAdapter is the base class for change point models adapters."""
+
+    @abstractmethod
+    def get_change_points_intervals(self, df: pd.DataFrame, in_column: str) -> List[TTimestampInterval]:
```

Comment:
Let there be 3 methods:
1. `_build_intervals` -- already implemented
2. `get_change_points(self, df: pd.DataFrame, in_column: str) -> List[pd.Timestamp]` -- abstract method
3. `get_change_points_intervals(self, df: pd.DataFrame, in_column: str) -> List[TTimestampInterval])` -- get the change points using `get_change_points` method and create intervals using `_build_intervals`